service/pkg/db/marshalHelpers.go

```diff
@@ -122,3 +124,65 @@ func KasKeysProtoJSON(keysJSON []byte) ([]*policy.KasKey, error) {
 	}
 	return keys, nil
 }
+
+func formatAlg(alg policy.Algorithm) (string, error) {
+	switch alg {
+	case policy.Algorithm_ALGORITHM_RSA_2048:
+		return "rsa:2048", nil
+	case policy.Algorithm_ALGORITHM_RSA_4096:
+		return "rsa:4096", nil
+	case policy.Algorithm_ALGORITHM_EC_P256:
+		return "ec:secp256r1", nil
+	case policy.Algorithm_ALGORITHM_EC_P384:
+		return "ec:secp384r1", nil
+	case policy.Algorithm_ALGORITHM_EC_P521:
+		return "ec:secp512r1", nil
+	case policy.Algorithm_ALGORITHM_UNSPECIFIED:
+		fallthrough
+	default:
+		return "", fmt.Errorf("unsupported algorithm: %s", alg)
+	}
+}
+
+func UnmarshalDefaultKasKey(keysJSON []byte, key *kasregistry.DefaultKasKey) error {
+	if keysJSON != nil {
+		if err := protojson.Unmarshal(keysJSON, key); err != nil {
+			return err
+		}
+
+		alg, err := strconv.ParseInt(key.GetPublicKey().GetAlgorithm(), 10, 32)
+		if err != nil {
+			return err
+		}
+		key.PublicKey.Algorithm, err = formatAlg(policy.Algorithm(alg))
+		if err != nil {
+			return err
+		}
+		// Base64 decode the public key
+		pem, err := base64.StdEncoding.DecodeString(key.GetPublicKey().GetPem())
+		if err != nil {
+			return err
+		}
+		key.PublicKey.Pem = string(pem)
```

Comment:
Maybe try not to partially update the value on error?
```suggestion
		algEnum, err := formatAlg(policy.Algorithm(alg))
		if err != nil {
			return err
		}
		// Base64 decode the public key
		pem, err := base64.StdEncoding.DecodeString(key.GetPublicKey().GetPem())
		if err != nil {
			return err
		}
		key.PublicKey.Algorithm = algEnum
		key.PublicKey.Pem = string(pem)
```